File: src/learning_topic/scripts/person_publisher.py
#! /usr/bin/env python3
# -*- coding: utf-8 -*-
import rospy
from learning_topic.msg import Person


def person_publisher():

    rospy.init_node("person_publisher", anonymous=True)

    person_info_pub = rospy.Publisher("/person_info", Person, queue_size=10)

    rate = rospy.Rate(10)

    while not rospy.is_shutdown():
        person_msg = Person()
        person_msg.name = "Tomi"
        person_msg.age = 10
        person_msg.sex = Person.male

        person_info_pub.publish(person_msg)
        rospy.loginfo(
            "Publisher person message [%s, %d, %d]",
            person_msg.name,
            person_msg.age,
            person_msg.sex,
        )

        rate.sleep()


if __name__ == "__main__":
    try:
        person_publisher()
    except rospy.ROSInterruptException:
        pass

File: src/learning_topic/src/person_publisher.cpp
#include <ros/ros.h>
#include "learning_topic/Person.h"

/* 
*/

int main(int argc, char **argv)
{
    //init
    ros::init(argc, argv, "Person_publisher");
    ros::NodeHandle n;

    ros::Publisher person_info_pub = n.advertise<learning_topic::Person>("/person_info", 10);

    ros::Rate loop_rate(1);

    int count = 0;
    while (ros::ok())
    {
        learning_topic::Person person_msg;
        person_msg.name = "Tom";
        person_msg.age = 10;
        person_msg.sex = learning_topic::Person::male;

        person_info_pub.publish(person_msg);
        ROS_INFO("Publish Person Info: name: %s age: %d sex: %d",
                 person_msg.name.c_str(), person_msg.age, person_msg.sex);

        loop_rate.sleep();
    }
}